### PR TITLE
Expose visitor through advanced namespace.

### PR DESCRIPTION
**Pull request by JimBobSquarePants**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
- Adds new `AcceptVisitor` extension method to the `Advanced` namespace and redirects all calls to that method.

### CODE REVIEW

File: src/ImageSharp/Image.cs [hunk 1]
```diff
@@ -131,11 +130,12 @@ public Image<TPixel2> CloneAs<TPixel2>()
...
 
         /// <summary>
-        /// Accept a <see cref="IImageVisitor"/>.
+        /// Accepts a <see cref="IImageVisitor"/>.
         /// Implemented by <see cref="Image{TPixel}"/> invoking <see cref="IImageVisitor.Visit{TPixel}"/>
         /// with the pixel type of the image.
         /// </summary>
-        internal abstract void AcceptVisitor(IImageVisitor visitor);
+        /// <param name="visitor">The visitor.</param>
+        protected internal abstract void Accept(IImageVisitor visitor);
```
Comment (antonfirsov) [inline src/ImageSharp/Image.cs]: It's unlikely that anyone will try to subclass `Image`, but in theory, we expose an extension point with `protected internal`, so `internal` is probably better.

File: src/ImageSharp/Advanced/AdvancedImageExtensions.cs [hunk 2]
```diff
@@ -15,6 +15,15 @@ namespace SixLabors.ImageSharp.Advanced
     /// </summary>
     public static class AdvancedImageExtensions
     {
+        /// <summary>
+        /// Accepts a <see cref="IImageVisitor"/> to implement a double-dispatch pattern in order to
+        /// apply pixel-specific operations on non-generic <see cref="Image"/> instances
+        /// </summary>
+        /// <param name="source">The source.</param>
+        /// <param name="visitor">The visitor.</param>
+        public static void AcceptVisitor(this Image source, IImageVisitor visitor)
```
Comment (antonfirsov) [inline src/ImageSharp/Advanced/AdvancedImageExtensions.cs]: 👍

File: src/ImageSharp/Image.cs (hunk 1 above)
Comment (antonfirsov) [reply src/ImageSharp/Image.cs]: Ok, I've just seen #1011. See my comments there!

Comment (JimBobSquarePants) [reply src/ImageSharp/Image.cs]: Cool. Will update once I merge #1009

**Pull request merged**